components/feature/addons/src/main/java/mozilla/components/feature/addons/worker/Extensions.kt

```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons.worker
+
+import java.io.IOException
+import kotlinx.coroutines.CancellationException
+/**
+ * Indicates if an exception should be reported to the crash reporter.
+ */
+internal fun Exception.shouldReport(): Boolean = cause is IOException || this is CancellationException
```

Comment:
[ JobCancellationException is an internal class](https://github.com/Kotlin/kotlinx.coroutines/blob/3bb3e55a53359ebe268c7def40c6b016ca0e5f9a/kotlinx-coroutines-core/jvm/src/Exceptions.kt#L29) for this reason we have to use it parent class `CancellationException`